File: task3/Ruby/Subtask4.rb
def print_diamond(n)
  diamond = ""

  # Upper part of the diamond
  (0...n).each do |i|
    diamond += " " * (n - i - 1) + "*" * (2 * i + 1) + "\n"
  end

  # Lower part of the diamond
  (n - 2).downto(0) do |i|
    diamond += " " * (n - i - 1) + "*" * (2 * i + 1) + "\n"
  end

  diamond
end

begin
  # Open the input file and read the number
  input_file = 'vsofficial.txt'
  n = File.read(input_file).to_i

  # Generate the diamond pattern
  diamond = print_diamond(n)

  # Write the diamond pattern to the output file
  output_file = 'output.txt'
  File.write(output_file, diamond)

  puts "Diamond pattern successfully written to #{output_file}."
rescue => e
  # Handle any potential errors (e.g., file not found)
  puts "An error occurred: #{e.message}"
end

File: task3/GO/Subtask4.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"strconv"
)

func main() {
	// Open the input file for reading
	inputFile, err := os.Open("vsofficial.txt")
	if err != nil {
		fmt.Println("Error opening input file:", err)
		return
	}
	defer inputFile.Close()

	// Read the number from the input file
	scanner := bufio.NewScanner(inputFile)
	if !scanner.Scan() {
		fmt.Println("Error reading number from input file")
		return
	}
	n, err := strconv.Atoi(scanner.Text())
	if err != nil {
		fmt.Println("Error converting input to integer:", err)
		return
	}

	// Open the output file for writing
	outputFile, err := os.Create("output.txt")
	if err != nil {
		fmt.Println("Error creating output file:", err)
		return
	}
	defer outputFile.Close()

	// Generate the diamond pattern and write to the output file
	printDiamond(n, outputFile)
}

// Function to print the diamond pattern
func printDiamond(n int, file *os.File) {
	// Upper part of the diamond
	for i := 0; i < n; i++ {
		for j := 0; j < n-i-1; j++ {
			file.WriteString(" ")
		}
		for j := 0; j < 2*i+1; j++ {
			file.WriteString("*")
		}
		file.WriteString("\n")
	}

	// Lower part of the diamond
	for i := n - 2; i >= 0; i-- {
		for j := 0; j < n-i-1; j++ {
			file.WriteString(" ")
		}
		for j := 0; j < 2*i+1; j++ {
			file.WriteString("*")
		}
		file.WriteString("\n")
	}
}
